cli/src/pcluster/templates/awsbatch_builder.py

```diff
@@ -151,6 +151,19 @@ def _add_resources(self):
         self._code_build_notification_rule = self._add_code_build_notification_rule()
         self._manage_docker_images_custom_resource.add_depends_on(self._code_build_notification_rule)
 
+    def _launch_template(self):
+        launch_template = ec2.CfnLaunchTemplate(
+            self.stack_scope,
+            "PclusterComputeEnvironmentLaunchTemplate",
+            launch_template_data=ec2.CfnLaunchTemplate.LaunchTemplateDataProperty(
+                metadata_options=ec2.CfnLaunchTemplate.MetadataOptionsProperty(http_tokens="required")
```

Comment:
Unit tests should cover the setting of this metadata.

Reply:
At this moment afaik we do these tests by asserting on the produced Cfn we generate with CDK (see `test_required_imds_v2`)